server/implementation/src/main/java/io/smallrye/graphql/execution/datafetcher/helper/ReflectionHelper.java

```diff
@@ -38,6 +41,39 @@ public ReflectionHelper(Operation operation, EventEmitter eventEmitter) {
         this.method = lookupMethod(operationClass, operation);
     }
 
+    public <T> T invokePrivileged(Object... arguments) {
+        final ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+        return invokePrivileged(tccl, arguments);
+    }
+
+    public <T> T invokePrivileged(final ClassLoader classLoader, Object... arguments) {
+
+        try {
+            return (T) AccessController
+                    .doPrivileged(new PrivilegedExceptionAction<Object>() {
+                        @Override
+                        public Object run() {
+                            ClassLoader originalTccl = Thread.currentThread()
+                                    .getContextClassLoader();
+                            Thread.currentThread().setContextClassLoader(classLoader);
+
+                            try {
+                                return invoke(arguments);
+                            } catch (Exception ex) {
+                                // TODO: Handle this better.
+                                throw new RuntimeException(ex);
+                            } finally {
+                                if (originalTccl != null) {
+                                    Thread.currentThread().setContextClassLoader(classLoader);
+                                }
```

Comment:
```suggestion
                                Thread.currentThread().setContextClassLoader(classLoader);
```

I would suggest always setting it back to the original loader - even if it was null.  A null classloader (as weird as it sounds) actually has a meaning - it is the JDK's bootstrap classloader.  I can't imagine a case where the thread's context classloader would be null before coming here, but it's always better to set it back to what it was before.